lib/facter/ec2.rb

```diff
@@ -46,8 +46,15 @@
     end
 
     setcode do
-      @querier.fetch
+      userdata = @querier.fetch
+      begin
+        userdata.encode(Encoding::UTF_8)
+      rescue EncodingError
```

Comment:
yes, I just saw how it's currently working here : https://github.com/puppetlabs/facter/blob/2.x/lib/facter/util/normalization.rb#L47

Do you want me to do the same here ?